src/emcee/tests/unit/test_sampler.py

```diff
@@ -319,3 +325,50 @@ def test_walkers_independent_randn_offset_longdouble(nwalkers, ndim, offset):
         np.random.randn(nwalkers, ndim)
         + np.ones((nwalkers, ndim), dtype=np.longdouble) * offset
     )
+
+
+def test_sampler_seed():
+    nwalkers = 32
+    ndim = 3
+    nsteps = 25
+    np.random.seed(456)
+    coords = np.random.randn(nwalkers, ndim)
+    sampler1 = EnsembleSampler(nwalkers, ndim, normal_log_prob, seed=1234)
+    sampler2 = EnsembleSampler(nwalkers, ndim, normal_log_prob, seed=2)
+    sampler3 = EnsembleSampler(nwalkers, ndim, normal_log_prob, seed=1234)
+    sampler4 = EnsembleSampler(
+        nwalkers, ndim, normal_log_prob, seed=deepcopy(sampler1._random)
+    )
+    for sampler in (sampler1, sampler2, sampler3, sampler4):
+        sampler.run_mcmc(coords, nsteps)
+    for k in ["get_chain", "get_log_prob"]:
+        attr1 = getattr(sampler1, k)()
+        attr2 = getattr(sampler2, k)()
+        attr3 = getattr(sampler3, k)()
+        attr4 = getattr(sampler4, k)()
+        assert not np.allclose(attr1, attr2), "inconsistent {0}".format(k)
+        assert np.allclose(attr1, attr3), "inconsistent {0}".format(k)
+        assert np.allclose(attr1, attr4), "inconsistent {0}".format(k)
+
+
+def test_sampler_bad_seed():
+    nwalkers = 32
+    ndim = 3
+    with pytest.raises(TypeError, match="seed must be"):
+        EnsembleSampler(nwalkers, ndim, normal_log_prob, seed="bad_seed")
+
+@pytest.mark.skipif(
+        packaging.version.parse(np.__version__) < packaging.version.parse("1.17.0"),
+        reason="requires numpy 1.17.0 or higher",
+    )
+def test_sampler_generator():
```

Comment:
this test checks that you can use a Generator to seed the Sampler, but it doesn't prove that sampling is reproducible with it. I would reduce the number of steps to 5, create an `EnsembleSampler` with a seeded `Generator` and `run_mcmc` it. Then I would then create another EnsembleSampler with a `Generator` created with the same seed, and `run_mcmc` it. THen check that the output from the two sampling runs are identical.